### PR TITLE
Append Kibana Security APIs

### PR DESCRIPTION
**Pull request by stefnestor**
@pmuellr @jportner to better introspect Kibana Security/Space issues, adds APIs into Kibana diagnostic: 
- [List Kibana Roles](https://www.elastic.co/guide/en/kibana/current/role-management-api-get.html)
- [List Kibana Spaces](https://www.elastic.co/guide/en/kibana/current/spaces-api-get-all.html)
    - Enables some introspection to mitigate [this](https://github.com/elastic/support-diagnostics/issues/578)'s impact.
- Get Current Kibana User
    - URL marks internal, but this is documented [here](https://github.com/elastic/support-diagnostics/issues/479#issuecomment-825148583) & is frequently used to diagnose Kibana permission issues.

Kindly review
1. If endpoints return user private/sensitive data.
2. Initial available versions.
    - I pulled via Github search / oldest doc, but there's been a lot of migrations to pinpoint oldest well.

cc: @pickypg

### CODE REVIEW

**Comment by pmuellr** [comment]
I'm just seeing a single commit 7cd356748db12c84ddce6c5b9f219186b47b2665 with a change to a yaml file.  Is there supposed to be some code here as well?

**Comment by jportner** [comment]
* **List Kibana roles**: This should be OK, it doesn't return private data. This was introduced in #19723, initial available version is 6.4.0.
* **List Kibana spaces**: This should be OK but it's not clear to me how having this information will assist in troubleshooting the linked issue. Can you clarify what the expectation here is?
* **Get Current Kibana User**: This can contain some "private data" in terms of personal information, example:
   ```json
   {
     "username": "elastic",
     "roles": [
       "superuser"
     ],
     "full_name": null,
     "email": null,
     "metadata": {
       "_reserved": true
     },
     "enabled": true,
     "authentication_realm": {
       "name": "reserved",
       "type": "reserved"
     },
     "lookup_realm": {
       "name": "reserved",
       "type": "reserved"
     },
     "authentication_type": "realm",
     "authentication_provider": {
       "type": "basic",
       "name": "basic"
     }
   }
   ```
   If the user is authenticated via SSO and their ES is configured to map attributes correctly, the `full_name` and/or `email` fields can be populated. That said, the rest of this data is really valuable for troubleshooting, and I think we can justify collecting this small amount of personal information for a support case where we likely already have this info anyway.
   * `/internal/security/me`: this was introduced in #50695, initial available version is 7.6.0.
   * `/api/security/v1/me`: this API gives the same info, it is deprecated and is being replaced by the new API above; I believe it was introduced in https://github.com/elastic/x-pack/commit/003ef56664a74bbb31d23ada3f24895ff916502e, initial available version appears to be 5.0.0.

**Comment by pmuellr** [comment]
> List Kibana spaces: This should be OK but it's not clear to me how having this information will assist in troubleshooting the linked issue. Can you clarify what the expectation here is?

Currently, the diagnostics get the alerting rules and connectors from the default space only.  We'd like to have them for all spaces, so presumably the space names will end up showing up here.   The diagnostics code will eventually be changed to get the list of spaces, and for each one, get all the rules and connectors from each space.

**Comment by jportner** [comment]
> Currently, the diagnostics get the alerting rules and connectors from the default space only. We'd like to have them for all spaces, so presumably the space names will end up showing up here. The diagnostics code will eventually be changed to get the list of spaces, and for each one, get all the rules and connectors from each space.

Gotcha, that makes sense.

@stefnestor the List Kibana Spaces API was introduced in https://github.com/elastic/kibana/pull/21408, initial available version is 6.5.0.
However, the `include_authorized_purposes` parameter wasn't added until 7.11.0. I would omit that from your test script entirely, it won't give you useful information.

File: src/main/resources/kibana-rest.yml [hunk 1]
```diff
@@ -34,6 +34,14 @@ kibana_detection_engine_signals:
   versions:
     "> 7.10.0": "/api/detection_engine/prepackaged"
 
+kibana_roles:
+  versions:
+    ">= 6.4.0": "/api/security/role"
+
+kibana_spaces:
+  versions:
+    ">= 6.4.0": "/api/spaces/space?include_authorized_purposes=true"
```
Comment (jportner) [inline src/main/resources/kibana-rest.yml:43]:
```suggestion
    ">= 6.5.0": "/api/spaces/space"
```

File: src/main/resources/kibana-rest.yml [hunk 2]
```diff
@@ -45,3 +53,7 @@ kibana_stats:
 kibana_task_manager_help:
   versions:
     ">= 7.11.0": "/api/task_manager/_health"
+
+kibana_user:
+  versions:
+    ">= 7.6.0": "/internal/security/me"
```
Comment (jportner) [inline src/main/resources/kibana-rest.yml:59]:
```suggestion
    ">= 7.6.0": "/internal/security/me"
    ">= 5.0.0 < 7.6.0": "/api/security/v1/me"
```

**Comment by stefnestor** [comment]
Thanks, both 🎉 ! Late circling back, but changes accepted & ready for Dev approval for SupportOps to then merge. 🙏🏼

**Review by pickypg — APPROVED** [review]
LGTM

**Pull request merged**